src/connection.c

```diff
@@ -330,3 +328,24 @@ ConnectionGetOptionValue(PGconn *connection, char *optionKeyword)
 
 	return optionValue;
 }
+
+ShardId DoForAllShards(List* shardConnections, ShardAction action, void* arg)
```

Comment:
We find the failure semantics of the DoForAllShards function a bit confusing. Could you remove it and put the loops in each of the 3 tasks we want to perform?

In other words, you would get the following functions:

     static ShardId PgCopyPrepareTransactions(List *shardConnections);
     static ShardId PgCopyAbortTransactions(List *shardConnections);
     static ShardId PgCopyEndTransactions(List *shardConnections);